src/lib/sql/sql_pipeline_builder.hpp

```diff
@@ -35,12 +36,21 @@ class Optimizer;
  */
 class SQLPipelineBuilder final {
  public:
+  // Plan caches used if the respective cache is nullptr in this builder. Both default caches can be nullptr themselves.
```

Comment:
"if the respective cache is nullptr"? I think you mean the same, but "if with_(pqp|lqp)_cache are not used" would be clearer

Reply:
I changed it, but that's slightly less correct because if those are called with nullptrs the default cache is still used. So for a builder where you explicitly stated that you don't want to use caches, you still get a cache. Globals are great!

Do you want me to make this `std::optional<std::shared_ptr<SQLPhysicalPlanCache>>` to work around this?

Reply:
Why do we need an optional to work around this? In the constructor of the builder, set `_pqp_cache(default_pqp_cache)`. If `with_pqp_cache(nullptr)` is called, that is overwritten.

Reply:
ah, clever.